app/services/csv_exporter.rb

```diff
@@ -4,13 +4,17 @@ module CsvExporter
   def self.export(resources, columns, header = nil)
     header ||= default_header(columns)
     raise ArgumentError, "Columns and header row aren't the same length" unless columns.length == header.length
+    # we need to pass the current user to the enumerator which is executed by a seperate thread
```

Comment:
It took me some time to realize that although `resources` should already be filtered by the user (even though it's in another thread), the actual filter evaluation will be performed in this new thread, so it has to set the user correctly.

Could you please add this info to a comment?